internal/pkg/cli/pipeline_deploy.go

```diff
@@ -143,24 +149,41 @@ func newDeployPipelineOpts(vars deployPipelineVars) (*deployPipelineOpts, error)
 				},
 			}
 		},
+		wsAppName: wsAppName,
 		svcBuffer: &bytes.Buffer{},
 		jobBuffer: &bytes.Buffer{},
 	}, nil
 }
 
-// Validate returns an error if the flag values passed by the user are invalid.
+// Validate returns an error if the optional flag values passed by the user are invalid.
 func (o *deployPipelineOpts) Validate() error {
+	if o.wsAppName == "" {
+		return errNoAppInWorkspace
+	}
+	// The passed-in app name value must be the same as the workspace app name, as we need to be in the correct workspace to read the pipeline manifest.
+	if o.appName != "" && o.appName != o.wsAppName {
+			return fmt.Errorf("cannot specify app %s because the workspace is already registered with app %s", o.appName, o.wsAppName)
+	}
+	// Validate the app name.
+	if _, err := o.store.GetApplication(o.wsAppName); err != nil {
+		return fmt.Errorf("get application %s configuration: %w", o.wsAppName, err)
+	}
+	return nil
+}
+
+// Ask prompts the user for any unprovided required fields and validates them.
+func (o *deployPipelineOpts) Ask() error {
```

Comment:
thoughts on this instead? just to remove [the nesting](https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow)
```
if o.name == "" {
	if err := o.askPipelineName(); err != nil {
		return err
	}

	return nil
}

return o.validatePipelineName()
```

Reply:
Yes! Normally that would be much lovelier, but in this case we want to avoid validating the pipeline name that came from the selector because the user was presented with only valid options.

Reply:
i think my suggestion does avoid validating the pipeline name that comes from, although i just realized it could be simplified even more while still keeping the original logic:

```
if o.name == "" {
    return o.askPipelineName()
}

return o.validatePipelineName()
```

Reply:
Oh, sorry, right-- bc of the `return nil`. 

I'll also move the 'if' back to `validatePipelineName()`.